packages/console-extension/src/index.ts

```diff
@@ -405,6 +411,26 @@ async function activateConsole(
       void tracker.save(panel);
     });
 
+    if (options.subshell) {
+      panel.sessionContext.ready
+        .then(async () => {
+          // Ensure kernel has received kernel_info.
+          await panel.sessionContext.session!.kernel!.info;
+          const future =
+            panel.sessionContext.session!.kernel!.requestCreateSubshell({});
+          future.onReply = (
+            msg: KernelMessage.ICreateSubshellReplyMsg
+          ): void => {
+            const subshellId = msg.content.subshell_id;
+            panel.sessionContext.session!.kernel!.subshellId = subshellId;
+          };
+          await future.done;
+        })
+        .catch(reason => {
+          console.error('Failed to initialize SessionContext.', reason);
```

Comment:
I this the right messaging here? What if `sessionContext.ready` resolves ok but `.then()` throws (e.g. due to non-null assertions?). It would seem to me that the responsibility of this coude would be to let user know that subshells won't be available, not make a (possibly incorrect) statement about session context, right?

```suggestion
          console.error('Failed to initialize subshells.', reason);
```